modulo2/projeto-labefy/src/components/DetsPlaylist.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import axios from 'axios'
+import styled from 'styled-components'
+
+const headers = {
+  headers: {
+      Authorization: "Guilherme-Lira-Silveira"
+  }
+};
+
+const Audio = styled.iframe`
+  margin: 0 0;
+  text-align: center;
+  cursor: pointer;
+  height: 4vh;
+`;
+
+const CardUser = styled.div`
+    display: flex;
+    flex-direction: column;
+    width: 300px;
+    align-items: center;
+    justify-content: center;
+    margin: 10px;
+    border-radius: 40px;
+    border: 2px solid black;
+    height: 190px;
+    p {
+        font-weight: bolder;
+        font-size: medium;
+    }
+`
+const MusicContainer = styled.div`
+    display: flex;
+`
+
+export default class DetsPlaylist extends React.Component {
+  state = {
+    playlist: {},
+    songs: [],
+    page: "detsplaylist",
+    nameSong: "",
+    nameArtist: "",
+    nameUrl: "",
+    playlistId: ""
+  }
+
+onChangeNameSong = (e) => {
+    this.setState({ nameSong: e.target.value });
+};
+
+onChangeNameArtist = (e) => {
+  this.setState({ nameArtist: e.target.value });
+};
+
+onChangeNameUrl = (e) => {
+  this.setState({ nameUrl: e.target.value });
+};
+
+  getPlaylist = () => {
+    axios
+    .get(`https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${this.props.idPlaylist}/tracks`, headers)
+    .then((res) => this.setState({playlist: res.data.result.tracks}))
+    .catch((err) => console.log(err.response))
+  }
+
+  addTrackToPlaylist = () => {
+    const url = `https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${this.props.idPlaylist}/tracks`
+    const body = {nameSong: this.state.nameSong, nameArtist: this.state.nameArtist, nameUrl: this.state.nameUrl}
+  axios
+  .post(url, body, headers)
+  .then((res) => {
+    this.setState({nameSong: "", nameArtist: "", nameUrl: ""})
+    this.getgetPlaylist();
+    alert(`Música adicionada com sucesso!`)
+  })
+  .catch((err) => {
+    alert(err.response.data.message)
+    alert("Problema com a função: addTrackToPlaylist ")
+  })
+}
```

Comment:
Já na requisição de adicionar música para a playlist, na linha 69, você colocou as propriedades dentro do body diferente do que a documentação apresenta, deveria seguir nesse formato:
~~~Js
 const body = {
            name: this.state.nameSong, 
            artist: this.state.nameArtist,
            url: this.state.nameUrl
        }
~~~